destination/writer/writer.go

```diff
@@ -0,0 +1,172 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type Writer struct {
+	db    *mongo.Collection
+	table string
+}
+
+// Params is an incoming params for the NewWriter function.
+type Params struct {
+	DB    *mongo.Collection
+	Table string
+}
+
+// NewWriter creates new instance of the Writer.
+func NewWriter(ctx context.Context, params Params) (*Writer, error) {
+	writer := &Writer{
+		db:    params.DB,
+		table: params.Table,
+	}
+
+	return writer, nil
+}
+
+// InsertRecord inserts a sdk.Record into a Destination.
+func (w *Writer) Write(ctx context.Context, record sdk.Record) error {
+	if err := sdk.Util.Destination.Route(ctx,
+		record,
+		w.insert,
+		w.update,
+		w.delete,
+		w.insert,
+	); err != nil {
+		return fmt.Errorf("insert record error: %w", err)
+	}
+
+	return nil
+}
+
+func (w *Writer) Close(ctx context.Context) error {
+	err := w.db.Database().Client().Disconnect(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to close: %w", err)
+	}
+
+	return nil
+}
+
+const (
```

Comment:
please move constants to separate file or move to top on the  file

Reply:
fixed